server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -172,11 +176,42 @@ public final class IndexSettings {
     public static final Setting<Boolean> INDEX_SOFT_DELETES_SETTING = Setting.boolSetting(
         "index.soft_deletes.enabled",
         true,
+        new SoftDeleteValidator(),
         Property.IndexScope,
         Property.Final,
         Property.Deprecated
     );
 
+    private static class SoftDeleteValidator implements Setting.Validator<Boolean> {
```

Comment:
I think there are various downsides to this approach:

- We cannot set the setting anymore to false with 5.0, not even in tests which is still needed, since we still need to support all the translog related stuff for tables without soft-deletes until we removed in 6.0 So have to find a solution here. Because all these tests are failing now.
- The usecase is to make soft-deletes mandatory for newly created tables. This is depending on the index version. All this part of the meta data creation process which is handled in `MetadataCreateIndexService`.
- This is not an argument, but i had a peak at the other distributions and they also handle in MetadataCreateIndexService.

Reply:
I see, thanks for this clarification. So lets use the concrete validation.